ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManagerMapBuffer.java

```diff
@@ -405,10 +407,50 @@ public static long measureText(
             ? paragraphAttributes.getInt(PA_KEY_MAX_NUMBER_OF_LINES)
             : UNSET;
 
+      int numberOfLines =
+      paragraphAttributes.contains(PA_KEY_NUMBER_OF_LINES)
+        ? paragraphAttributes.getInt(PA_KEY_NUMBER_OF_LINES)
+        : UNSET;
+
+    int lines = layout.getLineCount();
+    if (numberOfLines != UNSET && numberOfLines != 0 && numberOfLines > lines && text.length() > 0) {
```

Comment:
Could we share this logic with TextLayoutManager? Or is this file already kind of just a copy of it?

Reply:
The whole file is a faster copy of TextLayoutManager that uses C++/Java shared buffer instead of using JNI.